topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -240,11 +240,12 @@ public void handle(Long oldTimerId) {
     }
 
     public void setupMetrics() {
-        new ClassLoaderMetrics().bindTo(METRICS_REGISTRY);
-        new JvmMemoryMetrics().bindTo(METRICS_REGISTRY);
-        new ProcessorMetrics().bindTo(METRICS_REGISTRY);
-        new JvmThreadMetrics().bindTo(METRICS_REGISTRY);
-        new JvmGcMetrics().bindTo(METRICS_REGISTRY);
+        this.metricsRegistry = (PrometheusMeterRegistry) BackendRegistries.getDefaultNow();
```

Comment:
interesting to know why this move should help :-)

Reply:
I knew it is something with metrics. So this was just a desperate move. But now we have the same approach in all operators.

Reply:
Let's see what will happen when I will remove this method in a coming PR. It's not needed for enabling these metrics. Let's hope it won't screw up tests again :-)

Reply:
The problem is not in the metrics but in the registry. It seems to have some kind of weird asynch initialization / destroying of the registry which is causing problem. Funny enough, I think we did the change which Stanislav changed now exactly because of NPEs on Jenkins :-o